### PR TITLE
[flutter_appauth] [flutter_appauth_interface] Add analysis_options in flutter_appauth_platform_interface

### PR DESCRIPTION
**Pull request by lsaudon**
- add [usePubspecOverrides in melos](https://melos.invertase.dev/configuration/overview#commandbootstrapusepubspecoverrides)
- Update dart and flutter
- add analysis_options in flutter_appauth_platform_interface
- remove unused class and mocktail

### CODE REVIEW

**Comment by MaikuB** [comment]
Apologies for getting to this late. Are you able to explain the reasons behind the changes in this PR? The main one that sticks out to me is the reason behind bumping the minimum Flutter and Dart SDK requirements. The changes in this PR don't seem to provide enough reason to bump the minimum requirements to limit who can use the plugin

**Comment by lsaudon** [comment]
I put back the old versions.

File: flutter_appauth_platform_interface/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# Changelog
```
Comment (MaikuB) [inline flutter_appauth_platform_interface/CHANGELOG.md:1]: can you please revert this change as it's not necessary and not even a convention being used within this repo. If you look at the various plugins in the community, you'll find this isn't added and in fact redundant on pub.dev when there's a "Changelog" tab with the same text as shown below

Screenshot 2022-12-03 at 9 11 35 pm

File: flutter_appauth_platform_interface/analysis_options.yaml [hunk 2]
```diff
@@ -0,0 +1,191 @@
+# Specify analysis options.
```
Comment (MaikuB) [inline flutter_appauth_platform_interface/analysis_options.yaml:1]: Thanks but it's not necessary to add this file here. I originally left this out due to issues back many years ago. It should be possible to just have this at the root of the repository. As you're trying to fix the linter, would you be able to try fix this up putting it at the root and remove all other instances of the file?

**Review by MaikuB — CHANGES_REQUESTED** [review]

File: flutter_appauth_platform_interface/CHANGELOG.md (hunk 1 above)
Comment (lsaudon) [reply flutter_appauth_platform_interface/CHANGELOG.md]: Done

File: flutter_appauth_platform_interface/analysis_options.yaml (hunk 2 above)
Comment (lsaudon) [reply flutter_appauth_platform_interface/analysis_options.yaml]: Done

Comment (MaikuB) [reply flutter_appauth_platform_interface/analysis_options.yaml]: Could you help remove the other instances of analysis_options.yaml whilst you're at it?

Comment (lsaudon) [reply flutter_appauth_platform_interface/analysis_options.yaml]: It's done, i remove in flutter_appauth, example and flutter_appauth_platform_interface.
and move in root of repository

File: flutter_appauth_platform_interface/pubspec.yaml [hunk 3]
```diff
@@ -5,14 +5,13 @@ homepage: https://github.com/MaikuB/flutter_appauth/tree/master/flutter_appauth_
...
 environment:
   sdk: ">=2.12.0 <3.0.0"
-  flutter: ">=1.10.0"
+  flutter: ">=1.12.13+hotfix.5"
 
 dependencies:
   flutter:
     sdk: flutter
-  plugin_platform_interface: ^2.0.0
+  plugin_platform_interface: ^2.1.3
```
Comment (MaikuB) [inline flutter_appauth_platform_interface/pubspec.yaml:13]: missed this but could you also revert this? this change isn't necessary as using caret syntax allows pub to resolve to this 2.1.3 if it's the latest version anyhow

Comment (lsaudon) [reply flutter_appauth_platform_interface/pubspec.yaml]: Done

**Review by MaikuB — APPROVED** [review]
